domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterControllerDebugImpl.kt

```
@@ -282,6 +285,91 @@ class PlatformParameterControllerDebugImpl @Inject constructor(
+        val result = loadEphemeralFeatureFlags().retrieveData()
+      ) {
+        is AsyncResult.Success -> {
+          val restoredValue = result.value
+            .firstOrNull { it.id == id }
+            ?.currentValue == true
+          AsyncResult.Success(restoredValue)
+        }
+        is AsyncResult.Failure -> {
+          AsyncResult.Failure(result.error)
+        }
+        is AsyncResult.Pending -> {
+          AsyncResult.Pending()
+        }
+      }
+    }
+  }
+
+  /**
+   * Resets the locally overridden platform parameter corresponding to the specified [id].
+   *
+   * This removes any locally overridden value for the specified platform parameter from the local
+   * override database and returns the currently active value from either the server or the default.
+   *
+   * @param id the [PlatformParameterId] of the platform parameter to reset
+   * @return a [DataProvider] that returns the current [PlatformParameterValue] for the given [id],
+   * or the default instance if no such parameter is found
+   */
+  fun resetPlatformParameter(id: PlatformParameterId): DataProvider<PlatformParameterValue?> {
+    return dataProviders.createInMemoryDataProviderAsync(
+      RESET_OVERRIDDEN_PLATFORM_PARAMETER_PROVIDER_ID
+    ) {
+      databaseStore.storeDataAsync(updateInMemoryCache = true) { oldDatabase ->
+        val updatedOverrides = oldDatabase.overriddenPlatformParameterList
+          .filterNot { it.id == id }
+
+        oldDatabase.toBuilder()
+          .clearOverriddenPlatformParameter()
+          .addAllOverriddenPlatformParameter(updatedOverrides)
+          .build()
+      }.await()
+
+      return@createInMemoryDataProviderAsync when (
+        val result = loadEphemeralPlatformParameters().retrieveData()
+      ) {
+        is AsyncResult.Success -> {
+          val restoredValue = result.value
+            .firstOrNull { it.id == id }
+            ?.currentValue ?: PlatformParameterValue.getDefaultInstance()
```

Can we just do

```kotlin
val restoredValue = result.value.first { it.id == id }.currentValue
```

since it’s safe to skip `?: PlatformParameterValue.getDefaultInstance()`? `currentValue` is guaranteed to be non-null, and the restored value should always be present in the list, so using `first { it.id == id }` also makes the code fail-fast if something unexpectedly goes wrong.

---

updated